lib/sparse/src/common/sparse_vector.rs

```diff
@@ -71,10 +74,12 @@ fn score_vectors<T: Ord + Eq>(
                 score += self_values[i] * other_values[j];
                 i += 1;
                 j += 1;
+                cpu_counter += 1;
```

Comment:
Putting the counter here would mean, that computing score between 2 huge vectors which have no intersection would require 0 cpu resources. I don't think this is correct, I would prefer to cost of this operations and just a sum: `self_indices.len() + other_indices.len()`